src/utils/numAbbr.js

```diff
@@ -0,0 +1,82 @@
+/**
+ * @flow
+ * @file Function to abbreviate a numberas a string in a locale-sensitive manner.
+ * @author Box
+ */
+
+import IntlMessageFormat from 'intl-messageformat';
+import { languages, numbers } from 'box-locale-data';
+
+interface NumAbbrOptions {
+    length: string;
+    locale: string;
+}
+
+export function numAbbrWithLocale(localedata: object, num: number, locale: string, options?: NumAbbrOptions): string {
```

Comment:
In normal usage, the caller would just call the `numAbbr` function to abbreviate the number in the current locale. They do not call `numAbbrWithLocale` directly.

But, for the unit tests, we needed a way to test other locales. So, I made this as a function that the unit tests can call where the localedata is passed in. In the unit tests (q.v.), we load the locale data from @box/cldr-data explicitly and pass them in. That way, we don't have to do any dynamic imports in the `numAbbr` code. We don't have to worry about the situation where the caller passes in the wrong locale spec compared to the localedata, because it is not a public API.

Reply:
Can you rename `localedata` to something like `numbersLocaleData`? My initial question was why can't you get `locale` from `localedata`. Now I realized that `localedata` here is just numbers.

Reply:
renamed